### PR TITLE
add autoload per PWAD feature

### PR DESCRIPTION
**Pull request by rfomin**
Add autoload from subdirectories with PWAD names. For example:
```
autoload/
├── aaliens.wad/
│   ├── aaliens_ws.wad
│   ├── umapinfo.lmp

```

### CODE REVIEW

File: prboom2/src/d_main.c [hunk 1]
```diff
@@ -1400,19 +1400,35 @@ static char *GetAutoloadDir(const char *iwadname)
...
     mkdir(result);
 #else
     mkdir(result, 0755);
 #endif
+    }
 
     return result;
 }
 
+const char *BaseName(const char *filename)
```
Comment (fabiangreffrath) [inline prboom2/src/d_main.c:1415]: Static, please.

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]

**Comment by fabiangreffrath** [comment]
Awesome, thank you! 

What's needed now is the ability to add a system-wide autoload directory on non-WIN32 systems, e.g. under /usr/share/prboom-plus/prboom-plus (this applies to Woof! as well, btw).

Then we could jnstall the UMAPINFO lump files for nerve.wad and masterlevels.wad there and finally remove all the hard-coded special treatment for NRFTL from the sources.

**Comment by rfomin** [comment]
> What's needed now is the ability to add a system-wide autoload directory on non-WIN32 systems, e.g. under /usr/share/prboom-plus/prboom-plus (this applies to Woof! as well, btw).

I thought this does it:
https://github.com/fabiangreffrath/woof/blob/49253820e3e28c8684194317e48b7b6d40874ef1/CMakeLists.txt#L255
I don't have non-WIN32 system, except maybe MSYS.

**Comment by fabiangreffrath** [comment]
But we currently only search for autoload subdirectories in the same directory where the config is saved.

**Comment by rfomin** [comment]
I've never run Doom outside of DOS/Win32, can't help here, sorry.

**Comment by kraflab** [comment]
This feature doesn't work with the launcher, as far as I can tell. The launcher has been a serious thorn in my side with heretic and it's showing up here as well.

Basically the launcher does some unexpected behaviour. Here's a possible scenario:

You have doom2, plutonia, and a pwad. When pr+ starts with no options, it picks doom2 because it's the first automatic match - _it loads this as the iwad_. There is no pwad. Then we hit the autoload code which autoloads the wad files for doom2. _Then_ the launcher runs. The user now selects plutonia and the pwad, which resets the wad file lists (removing the autoloads already done). Now the iwad is plutonia and there is a pwad. When you run the deh autoloading, you autoload for plutonia and the pwad.

**Pull request merged**